spyglass/src/main/java/com/linkedin/android/spyglass/mentions/MentionSpan.java

```diff
@@ -137,4 +139,85 @@ public void setSelectedTextColor(int color) {
     public void setSelectedBackgroundColor(int color) {
         SELECTED_BG_COLOR = color;
     }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeInt(NORMAL_TEXT_COLOR);
+        dest.writeInt(NORMAL_BG_COLOR);
+        dest.writeInt(SELECTED_TEXT_COLOR);
+        dest.writeInt(SELECTED_BG_COLOR);
+
+        dest.writeInt(getMention().getDeleteStyle().ordinal());
+        dest.writeInt(getDisplayMode().ordinal());
+        dest.writeString(getMention().getTextForDisplayMode(MentionDisplayMode.FULL));
+        dest.writeString(getMention().getTextForDisplayMode(MentionDisplayMode.PARTIAL));
+        dest.writeInt(isSelected() ? 1 : 0);
+        dest.writeInt(getMention().getId());
+        dest.writeString(getMention().getPrimaryText());
+    }
+
+    public MentionSpan(Parcel in) {
+        NORMAL_TEXT_COLOR = in.readInt();
+        NORMAL_BG_COLOR = in.readInt();
+        SELECTED_TEXT_COLOR = in.readInt();
+        SELECTED_BG_COLOR = in.readInt();
+
+        final Mentionable.MentionDeleteStyle deleteStyle = Mentionable.MentionDeleteStyle.values()[in.readInt()];
+        mDisplayMode = MentionDisplayMode.values()[in.readInt()];
+        final String fullDisplayModeText = in.readString();
+        final String partialDisplayModeText = in.readString();
+        setSelected((in.readInt() == 1));
+        final int id = in.readInt();
+        final String primaryText = in.readString();
+
+        mMention = new Mentionable() {
```

Comment:
I don't think this is correct behavior. When mentions are saved then restored, you are only restoring the properties associated with the Mentionable interface. If you were to call getMention() on a MentionSpan here, it would return this anonymous instance of the Mentionable interface and not the actual object which implements Mentionable, which is what I would expect.

Instead, I would suggest we have Mentionable extend Parcelable and allow users of the library to provide their own implementation for saving and restoring their models.